runelite-client/src/main/java/net/runelite/client/plugins/chathistory/ChatHistoryPlugin.java

```diff
@@ -167,6 +182,43 @@ public void onChatMessage(ChatMessage chatMessage)
 		}
 	}
 
+	@Subscribe
+	public void onMenuOpened(MenuOpened event)
+	{
+		if (event.getFirstEntry() == null || !config.copyToClipboard())
+		{
+			return;
+		}
+
+		final int groupId = TO_GROUP(event.getFirstEntry().getParam1());
+		final int childId = TO_CHILD(event.getFirstEntry().getParam1());
+
+		if (groupId != WidgetInfo.CHATBOX.getGroupId())
+		{
+			return;
+		}
+
+		final Widget widget = client.getWidget(groupId, childId);
+		final Widget parent = widget.getParent();
+		final Widget[] dynamicChildren = parent.getDynamicChildren();
+
+		// Convert current message static widget id to dynamic widget id of message node with message contents
+		final int dynamicChildId = (childId - 59) * 2 + 1;
```

Comment:
This feels way too much like it should be an API function than an, admittedly documented, magic number-ridden line of code in some plugin.

Reply:
I don't think this should be API method as I don't think it will be useful outside of this plugin. It's works just for getting chat message contents based on info from menu entry when menu is openened and nothing else.